units/src/amount/unsigned.rs

```diff
@@ -408,8 +403,10 @@ impl Amount {
 
     /// Converts to a signed amount.
     #[rustfmt::skip] // Moves code comments to the wrong line.
+    #[allow(clippy::missing_panics_doc)]
     pub fn to_signed(self) -> SignedAmount {
-        SignedAmount::from_sat_unchecked(self.to_sat() as i64) // Cast ok, signed amount and amount share positive range.
+        SignedAmount::from_sat(self.to_sat() as i64) // Cast ok, signed amount and amount share positive range.
+            .expect("range of Amount is within range of SignedAmount")
```

Comment:
Honestly, this is the kind of thing where I'd prefer `unchecked` (by turning it private instead of deleting). The compiler doesn't understand our invariant (even niche-filling optimization that I came up with cannot convey it perfectly, just some other upper bound), so this won't get optimized out in non-trivial code and it's the kind of thing where one wouldn't expect a performance hit.

But it's fine to leave it as-is for now. Even wait until someone complains.

Reply:
If you think the compiler won't understand this, we could do a match and an unsafe `core::hint::unreachable_unchecked`. I'd rather do that than introduce a `from_sat_unchecked` for a single call site.

Regardless, I think let's leave it for now.

Reply:
OK, that's a reasonable approach.